vulnerabilities/models.py

```diff
@@ -324,6 +324,20 @@ def get_or_create_from_purl(self, purl: PackageURL):
         ``purl`` PackageURL.
         """
         purl_fields = without_empty_values(purl.to_dict(encode=True))
+
+        # when there are 2 packages one with qualifiers and one without
+        # qualifiers, having all other fields same, this raises MultipleObjectsReturned
+        # so we are filling out the fields with empty value to avoid this
+        for field in PackageURL._fields:
```

Comment:
I am not sure I understand what you are trying to do with this change. Can you elaborate?

Reply:
when there are 2 packages one with qualifiers and one without qualifiers. For say `pkg:generic/postgres?foo=bar` and one package `pkg:generic/postgres`. So when we try to `get` second package using (type="generic", name="postgres"), it returns 2 packages which raises  `MultipleObjectsReturned`. By declaring qualifiers as an empty dict, we only get the package that doesn't have qualifiers

Reply:
@TG1999 this should not be part of this PR. And I am not sure that problem and the fix are correct